src/PostgREST/ApiRequest.hs

```diff
@@ -125,8 +126,11 @@ userApiRequest schema req reqBody
     case decodeContentType . fromMaybe "application/json" $ lookupHeader "content-type" of
       CTApplicationJSON ->
           either Left (\val -> case ensureUniform (pluralize val) of
-            Nothing -> Left "All object keys must match"
-            Just json -> Right json) (JSON.eitherDecode reqBody)
+              Nothing -> Left "All object keys must match"
+              Just json -> Right json)
+            (if BL.null reqBody && isTargetingProc
+              then Right emptyObject
+              else JSON.eitherDecode reqBody)
```

Comment:
there is also this which kind of looks the same, donno which is better
`note :: a -> Maybe b -> Either a b`